tests/qat/model/test_pydantic_hw_serialisation.py

```diff
@@ -0,0 +1,134 @@
+from __future__ import annotations
```

Comment:
Could the file names for tests mimic the structure of `src/qat/` (e.g. as unit tests so we can easily identify where tests should go)